algorithms/src/Kokkos_Sort.hpp

```diff
@@ -694,6 +711,9 @@ void sort(const Cuda& space,
 
 template <class ViewType>
 void sort(ViewType const& view) {
+  if (view.extent(0) == 0) {
+    return;
+  }
```

Comment:
Nope I think you need to fence before you exit early,

Reply:
Why?

Reply:
Similar to deep_copy.  Overload that does not take an execution space is synchronous.

Reply:
thanks for pointing this out, i think i fixed it. there was also another overload wihtout exespace agument to fix.